pkg/repository/prisma/dbsqlc/lease.sql

```diff
@@ -0,0 +1,61 @@
+-- name: GetLeasesToAcquire :exec
+SELECT
+    *
+FROM
+    "Lease"
+WHERE
+    "tenantId" = @tenantId::uuid
+    AND "kind" = @kind::"LeaseKind"
+    AND "expiresAt" < now()
+    AND "resourceId" = ANY(@resourceIds::text[])
+FOR UPDATE;
+
+-- name: AcquireLeases :many
+-- Attempts to acquire leases for a set of resources. Returns the acquired leases.
+INSERT INTO "Lease" (
+    "expiresAt",
+    "tenantId",
+    "resourceId",
+    "kind"
+)
+SELECT
+    now() + COALESCE(sqlc.narg('leaseDuration')::interval, '30 seconds'::interval),
+    @tenantId::uuid,
+    input."resourceId",
+    @kind::"LeaseKind"
+FROM (
+    SELECT
+        unnest(@resourceIds::text[]) AS "resourceId"
+    ) AS input
+-- On conflict, acquire the lease if the existing lease has expired.
+ON CONFLICT ("tenantId", "kind", "resourceId") DO UPDATE
+SET
+    "expiresAt" = EXCLUDED."expiresAt"
+WHERE
+    "Lease"."expiresAt" < now() OR
```

Comment:
Intended OR?
`Returns the renewed leases.` will always update on conflict for existingLeaseIds even if not expired?

Reply:
Yes, I actually merged the renewal and acquire logic into a single query, so it's expected that it renews the lease as well. Deleting the old RenewLeases queries since they're unused.